application/src/main/java/org/thingsboard/server/service/subscription/DefaultTbEntityDataSubscriptionService.java

```diff
@@ -564,13 +564,8 @@ private ListenableFuture<TbEntityDataSubCtx> handleGetTsCmd(TbEntityDataSubCtx c
                         for (ReadTsKvQueryResult queryResult : queryResults) {
                             String queryKey = queriesKeys.get(queryResult.getQueryId());
                             if (queryKey != null) {
-                                TsValue[] tsValues = entityData.getTimeseries().get(queryKey);
-                                if (tsValues == null) {
-                                    tsValues = queryResult.toTsValues();
-                                } else {
-                                    tsValues = ArrayUtils.addAll(tsValues, queryResult.toTsValues());
-                                }
-                                entityData.getTimeseries().put(queryKey, tsValues);
+                                entityData.getTimeseries().merge(queryKey, queryResult.toTsValues(), ArrayUtils::addAll);
+                                lastTsMap.merge(queryKey, queryResult.getLastEntryTs(), Math::max);
```

Comment:
When `fetchLatestPreviousPoint` is true, the second `queryResult` will be for latestPreviousPoint.
Add to lastTsMap max value between the first and the second query's `lastEntryTs`.